daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/ImmArray.scala

```diff
@@ -72,7 +72,7 @@ final class ImmArray[+A] private (
   private def uncheckedGet(idx: Int): A = array(start + idx)
 
   /** O(1) */
-  def length: Int = len
+  lazy val length: Int = len
```

Comment:
because ImmArray array is immutable there is no point in calculating length each time this function is called